topi/include/topi/transform.h

```diff
@@ -1247,5 +1247,31 @@ inline Tensor ndarray_size(const Tensor& src,
   }, name, tag);
 }
 
+/*!
+ * \brief Returns a one-hot tensor where the locations repsented by indices take value 1, 
+    other locations take value 0.
+ * \param indices locations to set to 1.
+ * \param depth depth of the one-hot dimension.
+ * \param name output tensor name.
+ * \param tag output tensor tag.
+ * \return one-hot tensor.
+ */
+inline Tensor one_hot(const Tensor& indices,
+                      int depth,
+                      const std::string name = "T_one_hot",
+                      const std::string tag = kInjective) {
+  Array<Expr> out_shape = indices->shape;
+  out_shape.push_back(depth);
+  return compute(out_shape, [&](const Array<Var>& iter_vars) {
+    Array<Var> outer_indices;
+    for (size_t i = 0; i < iter_vars.size() - 1; i++) {
+      outer_indices.push_back(iter_vars[i]);
+    }
+
+    auto idx = iter_vars[iter_vars.size() - 1];
+    return tvm::if_then_else(indices(outer_indices) == idx, 1, 0);
```

Comment:
You should use `ir::Select::make` here for better performance

Reply:
Do you require indices to be int type? Otherwise, this line could fail if indices.dtype is float32 without type cast.
You should add a test case for indices with float32 type.

Reply:
Can you expand on how to use ir::Select::make here? And yes, indices must be an int type.

Reply:
Just replace `tvm::if_then_else` by `ir::Select::make`. You can find use case at https://github.com/dmlc/tvm/blob/master/topi/include/topi/nn/pooling.h#L265.

Reply:
Updated